packages/flame/lib/src/rendering/shadow3d_decorator.dart

```diff
@@ -117,10 +120,18 @@ class Shadow3DDecorator extends Decorator {
     _paint = null;
   }
 
+  /// Shadow's base color before opacity. This defaults to pitch-black.
+  Color get baseColor => _baseColor;
+  Color _baseColor;
+  set baseColor(Color value) {
+    _baseColor = value;
+    _paint = null;
+  }
+
   Paint? _paint;
   Paint _makePaint() {
     final paint = Paint();
-    final color = Color.fromRGBO(0, 0, 0, opacity);
+    final color = baseColor.withAlpha((opacity * 255).toInt());
```

Comment:
There is an extension on Color in Flame that does this that can be used instead.

Reply:
`baseColor.withValues(alpha: opacity)`?

Reply:
Ah, it wasn't an extension, it was built-in:
https://api.flutter.dev/flutter/dart-ui/Color/withOpacity.html

Reply:
![Screenshot 2024-12-02 at 3 07 52 PM](https://github.com/user-attachments/assets/55ffb202-cb13-4dab-ae0a-6bb8c2be26db)

looks like withValues is the new withOpacity? but either way, much cleaner

Reply:
@spydon it looks like it might be the new withOpacity, but it might only be on the beta channel right now, so let me get that aimed back at main Flutter channel and plop withOpacity back in.